actionview/lib/action_view/template/resolver.rb

```diff
@@ -19,7 +19,7 @@ class PathParser # :nodoc:
       def build_path_regex
         handlers = Template::Handlers.extensions.map { |x| Regexp.escape(x) }.join("|")
         formats = Template::Types.symbols.map { |x| Regexp.escape(x) }.join("|")
-        locales = "[a-z]{2}(?:-[A-Z]{2})?"
+        locales = "[a-z]{2}(?:[-_][A-Z]{2})?"
```

Comment:
Why don't make this regex configurable? For example, we use a non-standard locale "global", it does not match this regular expression.

Reply:
Can you clarify why `global` is desirable over the default of not specifying a locale? I don't think that's a pattern we've seen or have tests for. Do you have `:global` configured in `I18n.available_locales`?

Reply:
I understand that `rails-i18n` only supports ISO codes listed [here](https://github.com/svenfuchs/rails-i18n#available-locales) but I haven't tried anything else. There could be a way to get around that. For Ruby `i18n`, RFC4646 and RFC4647 compliance doesn't seem to be [enforced](https://github.com/ruby-i18n/i18n#pluggable-features).

For your specific use case, can you explain the rationale behind using a non-standard locale `global` vs the standard which accepts a language and a region code? `en-US` for US users, `en` for all other english speaking users. That communicates that english the language in both cases, but you have specific needs for the US region.

If non-standard locales need to be supported for backward compatibility reasons, my impression is that using `I18n` config to build the regex would be safer than allowing a configurable regex, but there could be a performance tradeoff. Also, a refactor might be needed to avoid hacking the details back to the build regex method. Of course this is just an opinion, I don't have the full picture to make this call.

Reply:
global name selected exactly for countries grouping and we didn't want to deal with region that time. having only locale seemed  very simple that time. also in my opinion different regions like en-us en-gb means different yml files but in our case global=en and only haml templates are different.

Reply:
I think the problem is that the behavior that anything you config in `I18n.available_locales` would work (intended or unintended) was working before. 

And in Rails 7 that behavior was removed. 
This PR doesn't provide that behavior. It just adds `lowdash` support which is POSIX locale format

Also I don't know about your  application @catz. But maybe you could make it work if you use [request `variant`](https://guides.rubyonrails.org/layouts_and_rendering.html#the-variants-option)  `global` 

I think that making the REGEX configurable has a lot of impact. 
Maybe we should make that considers the I18n  `available_locales` configuration ?

Reply:
@jhawthorn any thoughts on this discussion? I can modify this PR if accepting any `I18n` config is desired. I'll need some direction in that case as some decisions were probably taken when the code was changed.

Reply:
Thank you for the idea, @jvillarejo! request variant is suitable for our needs.

Reply:
I suspect if we use _only_ `I18n. available_locales` we'd likely have the same type of issue, with folks who use AV locals which aren't configured as available.

I've merged this PR (thank you!) but if the lack of support is causing hardship would definitely consider a PR to support any available_locale OR the existing format.

Reply:
I actually encountered the same problem when trying to update an app to rails 7 that uses non-standard locale names. In this case it is because of multi-tenancy: The app hosts several tenants and uses i18n for per-tenant text changes and (sometimes) per-tenant templates.

Instead of burying this in an old discussion, I opted to implement the suggestion with `I18n.available_locales` here: https://github.com/rails/rails/pull/45169

Any other solution that let's me keep the locale names would be appreciated just as well.